source/particles/particle_handler.cc

```diff
@@ -2222,13 +2286,13 @@ namespace Particles
                             // redo the loop; otherwise move on to next particle
                             if (child_index != 0)
                               {
-                                particles[child->active_cell_index()].push_back(
-                                  particles[stored_index][i]);
+                                insert_particle(cache->particles[i],
+                                                child,
+                                                owned_particles);
 
-                                particles[stored_index][i] =
-                                  particles[stored_index].back();
-                                particles[stored_index].resize(
-                                  particles[stored_index].size() - 1);
+                                cache->particles[i] = cache->particles.back();
+                                cache->particles.resize(
+                                  cache->particles.size() - 1);
```

Comment:
Assuming we move all particles away from child(0) this will end with an empty but valid vector in the cache for child(0). Is this equivalent to what happens to other cells that have no particles?
Also we may need to erase the list entry in `owned_particles` if that happens. Maybe we should replace these lines with a call to `remove_particle()`?

Reply:
I thought about it, but it gets more complicated, because we would need to take care of the handle as well (setting it to invalid before we enter `remove_particle()`). And then we need to run the `i` loop in reverse order, as we should not remove the particle as we still keep the index `i` into the array. So for now, I added the step to delete the cell from the list instead.